src/geode/mesh/helpers/rasterize.cpp

```diff
@@ -705,41 +711,88 @@ namespace
                         { ( j + 0.5 ) * grid_.cell_length_in_direction( 1 ),
                             ( k + 0.5 ) * grid_.cell_length_in_direction( 2 ) }
                     };
-                    if( !is_jk_to_process( triangle, point, j, k ) )
+                    const auto jk_process =
+                        is_jk_to_process( triangle, point, j, k );
+                    if( !jk_process.to_process )
                     {
                         continue;
                     }
 
-                    auto max_i =
-                        std::floor( projected_i_coordinate( normal,
-                                        points_[vertices_order_[0]], point )
-                                        / grid_.cell_length_in_direction( 0 )
-                                    - 0.5 );
-                    if( max_i < 0 )
-                    {
-                        max_i = 0;
-                    }
-                    values_[{ j, k }].emplace_back( max_i, counter_clockwise_ );
+                    auto i_coord = compute_i_coordinates(
+                        point, normal, jk_process.position );
+
+                    values_[{ j, k }].emplace_back(
+                        i_coord, counter_clockwise_ );
                 }
             }
         }
 
-        bool is_jk_to_process( const ProjectedTriangle& triangle,
+        geode::index_t compute_i_coordinates( const geode::Point2D& point,
+            const geode::Vector3D& normal,
+            geode::Position position ) const
+        {
+            std::array< geode::local_index_t, 3 > order{ 0, 1, 2 };
+            if( !counter_clockwise_ )
+            {
+                order = { 0, 2, 1 };
+            }
+            if( position == geode::Position::vertex0 )
```

Comment:
const auto floor_fabs = [](double value){return std::floor(std::fabs(value));};
Just a proposal to avoid writing the call of floor and fabs each time.
Simple lambda to replace it. As you want!